### PR TITLE
Upgrades to JDK8, JUnit 5 and Maven 3.1.1

### PR DESCRIPTION
**Pull request by gnodet**
This depends on https://github.com/apache/maven-doxia/pull/92

### CODE REVIEW

File: doxia-decoration-model/pom.xml [hunk 1]
```diff
@@ -36,16 +36,22 @@ under the License.
...
-      <groupId>org.codehaus.plexus</groupId>
-      <artifactId>plexus-component-annotations</artifactId>
+      <groupId>org.eclipse.sisu</groupId>
+      <artifactId>org.eclipse.sisu.plexus</artifactId>
+      <scope>compile</scope>
+    </dependency>
+    <dependency>
+      <groupId>javax.inject</groupId>
+      <artifactId>javax.inject</artifactId>
+      <version>1</version>
```
Comment (michael-o) [inline doxia-decoration-model/pom.xml:46]: Should this be provided?

**Review by michael-o — CHANGES_REQUESTED** [review]

File: doxia-decoration-model/pom.xml (hunk 1 above)
Comment (gnodet) [reply doxia-decoration-model/pom.xml:46]: I must admit I have quite a hard time to understand the `provided` scope which seems to depend on how you suppose the module will be used.  In this case, I don't know much about doxia, so I'll defer to you.  Fwiw, the plexus annotations were in the default scope, so I used the same.

Comment (michael-o) [reply doxia-decoration-model/pom.xml:46]: Let's leave it as-is for now.

File: doxia-integration-tools/pom.xml [hunk 2]
```diff
@@ -170,6 +166,24 @@
...
+      <groupId>org.codehaus.plexus</groupId>
+      <artifactId>plexus-testing</artifactId>
+      <version>1.1.0</version>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.apache.maven.wagon</groupId>
+      <artifactId>wagon-http-lightweight</artifactId>
+      <version>1.0</version>
+      <scope>test</scope>
```
Comment (michael-o) [inline doxia-integration-tools/pom.xml:186]: Why is this dep required when we have `wagon-http` in test?

**Review by michael-o — COMMENTED** [review]
I will merge this as soon as Doxia 2.0.0-M2 lands in Central.

**Review by michael-o — APPROVED** [review]

**Pull request closed without merging**